app/models/rent.rb

```diff
@@ -2,12 +2,19 @@ class Rent < ApplicationRecord
   belongs_to :user
   belongs_to :book
   validates :user, :book, :initial_date, :final_date, presence: true
+  validate :dates
+
+  protected
+
+  def dates
+    return if valid_dates?
+
+    errors.add(:base, 'The dates entered are invalid')
+    false
+  end
 
   def valid_dates?
-    book_rents = Rent.where('book_id = ?', book_id)
-    book_rents.delete(self)
-    book_rents.all? do |previous_rent|
-      previous_rent.initial_date > final_date || previous_rent.final_date < initial_date
-    end
+    book.rents.where(initial_date: initial_date..final_date,
+                     final_date: initial_date..final_date).empty?
```

Comment:
Ok, perfect. One question, this 2 condition works like a `OR` or a `AND`?
What do you need you?

Reply:
I just tested it and it works like an OR so it's exactly what I need. I'm gonna add one more validation because there's a use case that I missed.

Reply:
👍